fhirpipe/transform/fhir.py

```diff
@@ -1,122 +1,199 @@
+                    continue
+                # We retrieve all the attributes that start with the array path (with index)
+                attributes_in_array = {
+                    remove_root_path(path, position_ind + 1): attr
+                    for path, attr in path_attributes_map.items()
+                    if path.startswith(array_path)
+                }
+                # Build the array of sub fhir object
+                array = handle_array_attributes(attributes_in_array, row)
+                # Insert them a the right position
+                if array:
+                    insert_in_fhir_object(fhir_object, ".".join(split_path[: position_ind]), array)
+                arrays_done.add(array_path)
 
+    return fhir_object
 
-def rec_create_fhir_object(fhir_obj, structure, row, indices=[]):
-    if isinstance(structure, dict):
-        # If there are some inputs, we can build the objects to output
-        if "inputs" in structure.keys() and structure["inputs"]:
-            result = build_fhir_leaf_attribute(structure, row, indices)
-            fhir_obj[structure["name"]] = result
 
-        # Otherwise, we can recurse on children
+def fetch_values_from_dataframe(row, mapping_inputs, merging_script):
+    if len(mapping_inputs) == 1:
+        input = mapping_inputs[0]
+        if input["sqlValue"]:
+            sql = input["sqlValue"]
+            column_name = build_col_name(sql["table"], sql["column"], sql["owner"])
+            if input["script"]:
+                column_name = new_col_name(input["script"], column_name)
+            return row[column_name]
         else:
-            is_array = structure["fhirType"] == "array"
-
-            fhir_obj[structure["name"]] = [] if is_array else dict()
+            return input["staticValue"]
 
-            for attr in structure["children"]:
-                if is_array:
-                    create_fhir_list(fhir_obj, attr, structure["name"], row, indices)
-                else:
-                    rec_create_fhir_object(fhir_obj[structure["name"]], attr, row, indices)
+    else:
+        if not merging_script:
+            raise ValueError(
+                "You need to provide a merging script for attributes with several inputs."
+            )
+
+        cols_to_fetch = ([], [])
+        for input in mapping_inputs:
+            # Else retrieve the static value
+            if input["sqlValue"]:
+                sql = input["sqlValue"]
+                column_name = build_col_name(sql["table"], sql["column"], sql["owner"])
+                if input["script"]:
+                    column_name = new_col_name(input["script"], column_name)
+                cols_to_fetch[0].append(column_name)
+
+            # Else retrieve the static value
+            else:
+                cols_to_fetch[1].append(input["staticValue"])
+
+        return row[new_col_name(merging_script, cols_to_fetch)]
+
+
+def handle_array_attributes(attributes_in_array, row):
+    # Check lengths
```

Comment:
explain why we check array length and how we handle joined values as discussed together